### DISCUSSION TITLE
Will the development of openmmlab repos stop indefinitely?

### DISCUSSION THREAD

**Discussion by mchaniotakis** [open-mmlab/mmdetection#11815] [General]
I think that, even though openmmlab has a steep learning curve, once its set up is an amazing tool to work with. However the development seems to have stoped since the end of December after the passing of Professor Tang Xiaoou. Are there plans to continue the development in the future (or atleast assign new maintainers)?

**Comment by MiXaiLL76** [comment]
As an active user of mmdet (more than 100 projects launched on mmdet) I would like to be a maintainer of the project.
If this is possible, please let me know.

**Comment by davidpagnon** [comment]
Any news on this topic?

**Reply by MiXaiLL76** [reply]
no

**Comment by davidpagnon** [comment]
I have an answer: OpenMMLab is still maintaining these repositories but they are mostly shifting their efforts to LLMs: https://github.com/Tau-J/rtmlib/issues/36#issuecomment-2513517335

@MiXaiLL76: Maybe you could ask @Tau-J to advocate for you to be a maintainer?

**Comment by MiXaiLL76** [comment]
@hhaAndroid @hellock @ZwwWayne @yhcao6 @RangiLyu
Hi guys! I understand that you are very busy with LLM projects and have forgotten about this project. But it is used by a lot of people. Help us work, give people access to make changes or move the project to a public archive so that I can fork it completely and take it for personal support.

**Comment by maisonhai3** [comment]
They droped the MMLab.
MMLab's head professor passed years ago.
Then, they move staffs to InternLM.
Then, even the InternLM is half-dead now.

I love the MMLab works. Their code quality is greate. Easy to maintain.

**Comment by jacksteussie** [comment]
I wonder what future researchers will use from now on... vanilla pytorch? i guess they'll just continue to use mmlab until its not feasible to use with future versions of pytorch...

**Reply by MiXaiLL76** [reply]
> I wonder what future researchers will use from now on... vanilla pytorch? i guess they'll just continue to use mmlab until its not feasible to use with future versions of pytorch...

At the moment, I am already transferring projects to ultralytics, due to the impossibility of working with mmlab

**Reply by davidpagnon** [reply]
The problem with ultralytics is that there is a separate commercial license, so the products I release would not be fully open-source...

**Reply by mr-cn** [reply]
> > I wonder what future researchers will use from now on... vanilla pytorch? i guess they'll just continue to use mmlab until its not feasible to use with future versions of pytorch...
> 
> At the moment, I am already transferring projects to ultralytics, due to the impossibility of working with mmlab

I am in favor of ultralytics framework, too. Do you have any related document or blog to guide migrating to ultralytics?

**Comment by monkeycc** [comment]
Is there a similar recommendation framework

**Reply by MiXaiLL76** [reply]
> Is there a similar recommendation framework

https://github.com/ultralytics/ultralytics

**Reply by sirolf-otrebla** [reply]
@MiXaiLL76 it can't be used commercially.

**Reply by jacksteussie** [reply]
> @MiXaiLL76 it can't be used commercially.

i'm pretty sure it can?

**Reply by jacksteussie** [reply]
> @MiXaiLL76 it can't be used commercially.

From their website "Ultralytics offers cutting-edge software and AI models to drive innovation across industries. Whether you're a student, researcher, or enterprise, our flexible licensing options empower you to leverage Ultralytics YOLO models for research, proof of concept, development, or commercial use."

**Reply by davidpagnon** [reply]
It can be used commercially, but then the license is not free anymore

**Comment by sirolf-otrebla** [comment]
someone ought to fork mmdetection + mmengine at this point.

**Reply by mchaniotakis** [reply]
Well, some of the other repos are getting small updates. It's fine if mmdetection specifically is not updated this year (we can work around it, its in a good state) but we need to know it the development will ever pick up.

**Reply by sirolf-otrebla** [reply]
for me, for example, it's a problem I can make it work only w/ python 3.8. Python 3.8 is in an EOL state, with all that derives from it.

**Reply by mchaniotakis** [reply]
For me python 3.10 with onnxruntime 1.15  CUDA 11.8 torch 2.2 mmcv 2.1 mmengine 0.10.6 mmdet 3.2.0 works fine.

**Comment by sirolf-otrebla** [comment]
They forgot to add "but then you pay us" ;)

**Reply by MiXaiLL76** [reply]
If you are in regions where American laws do not apply, you can use. How will they reach you?)

For example, I am from Russia, how will the guys from the USA forbid me to use their library?)

**Comment by sirolf-otrebla** [comment]
As David says.

**Comment by MABatin** [comment]
I used mmdetection and mmsegmentation for my undergraduate research back in 2022. It's sad to see that it's basically dead now for so many days now.

**Comment by Akiy071** [comment]
真的很可惜看到官网已经停滞更新了

**Comment by lauriebax** [comment]
We've forked and updated 7 core repos (MMDetection, MMSegmentation, and others) to work with the latest PyTorch and made them production-ready for our AI platform.

We're committed to maintaining these repos going forward and encourage you to contribute features through pull requests.

Check them out if you need PyTorch compatibility: [https://github.com/VBTI-development](https://github.com/VBTI-development%E2%80%94feedback) —feedback welcome!

**Reply by mchaniotakis** [reply]
Excelent!

**Reply by sirolf-otrebla** [reply]
Great initiative, i am leaving you guys a star :)

**Reply by davidpagnon** [reply]
That's such a great news! Any plan to for MMpose, too? That's the main one I use :)

**Reply by matthimatik** [reply]
https://github.com/VBTI-development

**Reply by MatthiasHinz** [reply]
That sounds wonderful! I will try it.
